lib/git_worktree.rb

```diff
@@ -258,6 +258,14 @@ def checkout
     end
   end
 
+  def check_connection(url)
+    with_remote_options do |remote_options|
+      @repo.remotes
+           .create_anonymous(url)
+           .check_connection(:fetch, remote_options)
```

Comment:
`remote_options` here has our credentials and proxy info, so if that is configured, it will be used as it is with other methods in this class.

Reply:
Is check_connection here a rugged method?

Reply:
Yes, it is:

https://www.rubydoc.info/github/libgit2/rugged/Rugged%2FRemote:check_connection